### DISCUSSION TITLE
How to add trailing stop order?

### DISCUSSION THREAD

**Discussion by Fquant** [kernc/backtesting.py#686] [Q&A]
I want to add trailing stop order in the strategy, but I  don't know how to do.

**Comment by TarekVilela** [comment]
What I usually do is calculate the trailing stop indicator (ie. the n period trailing low) and make conditions inside the `next` function. Something like:

```
if position.is_long:
    if self.data['Close'][-1] <= self.buy_stop`:
        self.position.close()
```

**Reply by romanbsd** [reply]
You have to be careful with the data resolution for trailing orders. You might possibly want to use data.High or data.Low instead of a data.Close.

**Reply by TarekVilela** [reply]
Absolutely right. Especially if you're using the trailing low, otherwise you'll only stop if close == low.
